clients/src/test/java/org/apache/kafka/clients/consumer/KafkaConsumerTest.java

```diff
@@ -381,18 +377,23 @@ public boolean matches(ClientRequest request) {
             }
         }, new HeartbeatResponse(Errors.NONE.code()).toStruct(), coordinator);
 
+        // heartbeat interval is 2 seconds
+        time.sleep(heartbeatIntervalMs);
+        Thread.sleep(heartbeatIntervalMs);
```

Comment:
@guozhangwang How do you see that working? Would we pass an No-op Executor into AbstractCoordinator? Also, I guess we could move `ShutdownableThread` from connect into clients?